### PR TITLE
proper Results for Evaluator

### PR DESCRIPTION
**Pull request by philipcraig**
proper `Result` for `Evaluator`
proper `Err` for `Evaluator`
`evaluator_error_test_case` macro added and test cases created
100% code-coverage for `Evaluator`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,3 +13,4 @@ path = "src/main.rs"
 [dependencies]
 anyhow = "1.0.41"
 structopt = "0.3.21"
+thiserror ="1.0.26"
```
Comment (jbcoe) [inline Cargo.toml]: GitHub seems to think a newline is needed. I don't mind.

File: src/evaluator.rs [hunk 2]
```diff
@@ -136,7 +149,7 @@ mod tests {
         return_value: 101,
     }
 
-    evaluator_test_case! {
+    positive_evaluator_test_case! {
```
Comment (jbcoe) [inline src/evaluator.rs]: maybe stick with `evaluator_test_case` and use `evaluator_error_test_case`?

File: src/evaluator.rs [hunk 3]
```diff
@@ -146,11 +159,33 @@ mod tests {
...
         input: r#"
             let a = 42;
             a;"#,
         return_value: 0, // not 42
     }
+
+    macro_rules! negative_evaluator_test_case {
+        (name: $test_name:ident, input: $input:expr, return_value: $return_value:expr,) => {
+            #[test]
+            fn $test_name() {
```
Comment (jbcoe) [inline src/evaluator.rs:173]: can we get 'error' into the name somehow?

**Review by jbcoe — DISMISSED** [review]

File: Cargo.toml (hunk 1 above)
Comment (philipcraig) [reply Cargo.toml]: done

File: src/evaluator.rs (hunk 2 above)
Comment (philipcraig) [reply src/evaluator.rs]: Done

File: src/evaluator.rs (hunk 3 above)
Comment (philipcraig) [reply src/evaluator.rs:173]: Done

**Review by jbcoe — APPROVED** [review]

**Comment by jbcoe** [comment]
Thanks for this!

**Pull request merged**